ee/clickhouse/materialized_columns/columns.py

```diff
@@ -0,0 +1,90 @@
+import re
+from datetime import timedelta
+from functools import wraps
+from typing import Dict, no_type_check
+
+from django.utils.timezone import now
+
+from ee.clickhouse.client import sync_execute
+from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_DATABASE, TEST
+
+PropertyName = str
+ColumnName = str
+
+
+def cache_for(cache_time: timedelta):
+    def wrapper(fn):
+        @wraps(fn)
+        @no_type_check
+        def memoized_fn(*args, use_cache=not TEST):
+            current_time = now()
+            if (
+                args not in memoized_fn.__cache
+                or current_time - memoized_fn.__cache[args][0] > cache_time
+                or not use_cache
+            ):
+                memoized_fn.__cache[args] = (current_time, fn(*args))
+            return memoized_fn.__cache[args][1]
+
+        memoized_fn.__cache = {}
+        return memoized_fn
+
+    return wrapper
+
+
+@cache_for(timedelta(minutes=15))
+def get_materialized_columns(table: str) -> Dict[PropertyName, ColumnName]:
+    rows = sync_execute(
+        """
+        SELECT comment, name
+        FROM system.columns
+        WHERE database = %(database)s
+          AND table = %(table)s
+          AND default_kind = 'MATERIALIZED'
+          AND comment LIKE '%%column_materializer::%%'
```

Comment:
the comment begins with `column_materializer::` right? Any reason for the wildcard beforehand?

(Also, do double %% signify something else? Couldn't find anything about it in the CH documentation)

Reply:
This is escaping `%` in python strings. otherwise it thinks %c is some sort of a control sequence and errors. :)

Reply:
Got it, thanks!

One question still remains: Any specific reason for `'%%column_materializer::%%'` vs `'column_materializer::%%'` ?

Reply:
Oops, edit got lost. Nope, got used to escaping both sides when searching for queries :)